opendatahub-io/odh-dashboard
# Accelerator user flow

closes: https://github.com/opendatahub-io/odh-dashboard/issues/1440

## Description



- Added an accelerator profile CRD to track all accelerators
- Added accelerator profile user flow
   - accelerator dropdown
   - accelerator count UI
   - error/warning states
   - edit and create form business logic
   - hooks to manage accelerator and notebook/serving-runtime accelerator state
   - no longer use `api/gpu`
   - use accelerators in serving runtimes

## How Has This Been Tested?



1. apply the new crd to cluster
2. manually create a few CRDs - spec for what is needed.
4. edit one of the imagestreams in openshift to include the annotation (the key can be whatever you choose, make sure it matches one of the accelerator identifiers): `opendatahub.io/supported-accelerators: '{"nvidia.com/gpu: true"}'`
6. create a new workbench and select the image you changed
7. select the accelerator that has the matching identifier (should have a recommend label next to it)
8. should be no warning
9. open the dropdown for image stream again and see that the image you have selected has the recommended label
10. select a a different image stream and notice info message
11. reselect image that is recommended and select `None` for the accelerator (should see info message)
12. select a different accelerator that does not have a matching identifier (should see warning)
13. increase a decrease sizes (notice that any value over 2 gives a warning)
    - this is hard coded for now at 2 as detection is not done
14. submit form in a state that has an accelerator selected with values and non zero count
15. edit workbench and notice values are the same
16. set count to 0
17. this should remove the limit and resource counts for accelerator on the pod spec, not set to 0
18. do the same for serving runtimes

## Test Impact


none

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
